test/metabase/sync/sync_metadata/indexes_test.clj

```diff
@@ -55,3 +64,58 @@
                                                        :field-name "id"}))
             field-ids (#'sync.indexes/all-indexes->field-ids (:id (mt/db)) many-indexes)]
         (is (seq field-ids))))))
+
+(deftest sync-all-indexes!-test
+  (mt/test-drivers
+    (set/intersection (mt/normal-drivers-with-feature :index-info)
+                      (mt/normal-drivers-with-feature :describe-indexes)
+                      (mt/sql-jdbc-drivers))
+    (let [ds-to-index-def (mt/dataset-definition
+                           "ds_to_index"
+                           ["first_table"
+                            [{:field-name "first"
+                              :base-type :type/Integer}
+                             {:field-name "second"
+                              :base-type :type/Integer}
+                             {:field-name "third"
+                              :base-type :type/Integer}]
+                            [[1 2 3]]])]
+      (mt/dataset ds-to-index-def
+        (try
+          (testing "Base: Id is indexed"
+            (is (some? (t2/select-one :model/Field
+                                      {:where [:and
+                                               [:in :table_id (t2/select-fn-vec :id :model/Table :db_id (mt/id))]
+                                               [:= :display_name "ID"]
+                                               [:= :database_indexed true]]}))))
+          (testing "Base: Other columns have no index"
+            (let [other-fields (t2/select :model/Field
+                                          {:where [:and
+                                                   [:in :table_id (t2/select-fn-vec :id :model/Table :db_id (mt/id))]
+                                                   [:!= :display_name "ID"]]})]
+              (is (every? (comp (complement true?) :database_indexed) other-fields))
+              (is (= 3 (count other-fields)))))
+          (testing "All indexed fields picked up by sync (first, second, third)"
+            (doseq [field ["first" "second" "third"]
+                    :let [sql (sql.tx/create-index-sql driver/*driver* "first_table" [field])]]
+              (jdbc/execute! (sql-jdbc.conn/db->pooled-connection-spec (mt/db)) sql))
+            (binding [sync.indexes/*update-partition-size* 2]
+              (#'sync.indexes/sync-all-indexes! (mt/db)))
+            (is (every? :database_indexed
+                        (t2/select :model/Field
+                                   {:where [:in :table_id (t2/select-fn-vec :id :model/Table :db_id (mt/id))]}))))
+          (testing "Index removal is picked up correctly"
+            (doseq [field ["first" "second" "third"]
+                    :let [sql (format "DROP INDEX \"idx_first_table_%s\";" field)]]
```

Comment:
I think you probably want to use `sql.u/quote-name` or use honeysql, some drivers use backticks for quoting.